### PR TITLE
devicetree: Add DT_PROP_LAST macro

### PR DESCRIPTION
**Pull request by nordic-krch**
Add macro for getting the last element of an array type property.

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]

**Review by decsny — CHANGES_REQUESTED** [review]
The implementation is probably sound regarding this, but can you add a testcase for when the array length is 1, which is the main edge case

**Review by bjarki-andreasen — CHANGES_REQUESTED** [review]
Just use DT_PROP_BY_IDX(node, prop, UTIL_DEC(DT_PROP_LEN(node, prop))) :)

**Comment by nordic-krch** [comment]
@decsny test added.
@bjarki-andreasen changed to `UTIL_DEC`.

File: include/zephyr/devicetree.h [hunk 1]
```diff
@@ -898,6 +898,17 @@
...
+/**
+ * @brief Get the last element of an array type property
+ *
+ * @param node_id node identifier
+ * @param prop lowercase-and-underscores property name
+ *
+ * @return a representation of the last element of the property
+ */
+#define DT_PROP_LAST(node_id, prop) \
+	DT_PROP_BY_IDX(node_id, prop, UTIL_CAT(Z_UTIL_DEC_, DT_PROP_LEN(node_id, prop)))
```
Comment (gmarull) [inline include/zephyr/devicetree.h:910]: isn't this a layer violation? `Z_UTIL_DEC_` macros are impl details for `UTIL_DEC()`, why not use the public macro?

Comment (nordic-krch) [reply include/zephyr/devicetree.h]: It was already spotted by @bjarki-andreasen but I forgot to push the changes. It's updated now.

**Review by bjarki-andreasen — APPROVED** [review]
Looks good! Really fun timing since the reason I knew to use UTIL_DEC was because I needed this exact macro in a PR I'm working on :) now I can change it to DT_PROP_LAST()

**Review by decsny — APPROVED** [review]

**Review by rruuaanng — APPROVED** [review]

**Review by kartben — APPROVED** [review]
